src/client/nav/nav.js

```diff
@@ -1593,31 +1561,81 @@ spf.nav.createOptions_ = function(opt_options) {
 
 
 /**
- * @param {(Object|spf.nav.request.Info_)=} opt_info The request info object.
- * @return {spf.nav.request.Info_}
+ * Type definition for an object literal argument to {@link spf.nav.Info}.
+ *
+ * @typedef {{
+ *   current: (string|undefined),
+ *   history: (boolean|undefined),
+ *   original: (string|undefined),
+ *   position: (Array.<number>|undefined),
+ *   referer: (string|undefined),
+ *   reverse: (boolean|undefined),
+ *   type: (string|undefined)
+ * }}
  * @private
  */
-spf.nav.createInfo_ = function(opt_info) {
-  opt_info = opt_info || /** @type {spf.nav.request.Info_} */ ({});
-  var info = /** @type {spf.nav.request.Info_} */ ({
-    // The referer, stored in the history entry state object to allow the
-    // correct value to be sent to the server during back/forward.
-    // Compare against "undefined" to allow empty referer values in history.
-    referer: (opt_info.referer != undefined) ?
-                 opt_info.referer : window.location.href,
-    // The current URL, which will have already changed for history events.
-    // For this case, the opt_info.current value from the history state should
-    // be used instead.
-    current: (opt_info.history && opt_info.current) ?
-                 opt_info.current : window.location.href,
-    // Convert from undefined and other falsey values.
-    history: !!opt_info.history,
-    reverse: !!opt_info.reverse,
-    original: '',
-    position: opt_info.position || null,
-    type: opt_info.type || ''
-  });
-  return info;
+spf.nav.Info_;
+
+
+/**
+ * Data to track information about an SPF navigation.
+ *
+ * @param {(spf.nav.Info|spf.nav.Info_)=} opt_info A navigation info object.
+ * @constructor
+ * @struct
+ */
+spf.nav.Info = function(opt_info) {
+  opt_info = opt_info || /** @type {spf.nav.Info_} */ ({});
+  /**
+   * The current page URL. This differs from `referer` in that is always
+   * represents the current visible page regardless of history state.
+   * @type {string}
+   */
+  // The current URL will have already changed for history events, so for this
```

Comment:
Does mixing the two different styles of commenting work? Does the right type annotation get set?